.github/workflows/python_unit_test.yaml

```diff
@@ -36,6 +40,9 @@ jobs:
         uses: actions/setup-python@v4
         with:
           python-version: "3.6.7"
```

Comment:
We should be using python 3.9 in a lot of repos. Honestly we should probably find a way to matrix and run the tests against multiple versions. 3.6, 3.7, 3.8, 3.9, 3.10 and require 3.9 to pass

Reply:
So are you saying only 3.9 is the requirement, and the others are just to see what happens? Making 3.9 the only hard requirement might be tricky, but we should definitely be able to test across multiple versions in parallel.

Reply:
I'm not sure there's value in testing more versions if no one is checking the output - wouldn't we just be wasting computing resources?
I think it would be enough to default to 3.9 and allow overriding that

Reply:
I like that. That should be doable, I'll update the PR once its taken care of.

Reply:
I have updated this so it now calls "latest" which currently is 3.11 you can however specify per repo an input to force a specific version.